### PR TITLE
Fix #189, Virtualenv and Pipenv .gitignore support

### LINKED ISSUE #189: Virutualenv and Pipenv support within .gitignore
**Is your feature request related to a problem? Please describe.**
If you work with an IDE like PyCharm from Jetbrains you have to be careful when committing not to send the `.idea` folder by mistake because it is not listed in the `.gitignore`. Also missing is the exclusion of `venv` folders and `Pipenv` files in the `.gitignore`. I think some developers will be happy about this change.

**Describe the solution you'd like**
Structured .gitignore with the changes from above.

**Describe alternatives you've considered**
Do the excluding manually every time you have some of these in your code.

**Additional context**
I will add a PR that includes the above changes.

**Requester Info**
pavll

### PR DESCRIPTION
**Pull request by pavll**
Structured .gitignore

**Describe the contribution**
- Fixes #189 

**Testing performed**
 1. made a commit with the new .gitignore

**Contributor Info - All information REQUIRED for consideration of pull request**
Paul Oberosler, Individual

### CODE REVIEW

**Comment by jphickey** [comment]
I don't think this really needs to be in the project `.gitignore` file.  There are lots of different IDEs and they all leave different workspace/backup/etc files around.  While in the past some editor-specific patterns may have gotten introduced, in general each developer should just make their own ignore file (typically under `$HOME/.config/git/ignore`) that reflects the remnants left around by their editor/IDE of choice.

**Comment by pavll** [comment]
Ok, then I'll close this PR.

**Comment by jphickey** [comment]
> Ok, then I'll close this PR.

That was quick ... may want to get some other opinions on the matter before closing ... 

I'm not necessarily opposed to the venv/pycache stuff, since its likely that these files will get created no matter what IDE the user chooses.

**Comment by pavll** [comment]
Ok, I just thought that would take care of the whole .gitignore change thing. I can just throw out the IDE thing in a new commit and leave the venv thing in.

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,23 @@
...
+# Other stuff
 cmdUtil
 !/Subsystems/cmdUtil
+
+# cmake stuff
 CMakeFiles
 CMakeCache.txt
+
+# OS stuff
+.desktop
```
Comment (astrogeco) [inline .gitignore:22]: Remove .desktop

**Comment by astrogeco** [comment]
[CCB:2021-08-25](https://github.com/nasa/cFS/wiki/Community-CCB-Agenda:-August-25-2021) APPROVED
- remove .desktop from gitignore

**Review by astrogeco — APPROVED** [review]

**Pull request merged**